### PR TITLE
Fix forbidden apis on FIPS

### LINKED ISSUE #33179: fips builds don't pass forbidden api tests
Ever since https://github.com/elastic/elasticsearch/commit/82d10b484a20514e70f6d773544ffea62e57f711 the forbidden API checks have been failing on the fips builds. It isn't clear if this is a problem with the build or it is revealing a real problem with our fips compliance. I'm assigning to @atorok to investigate because he made the commit that broke the builds. If it turns out that this is a problem with our compliance he'll pass it off to someone who knows more about that stuff.

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java10,ES_RUNTIME_JAVA=java8fips,nodes=virtual&&linux/263/console

### PR DESCRIPTION
**Pull request by alpar-t**
- third party audit detects jar hell with JDK so we disable it
- jdk non portable in forbiddenapis detects classes being used from the
JDK ( for fips ) that are not portable, this is intended so we don't
scan for it on fips.
- different exclusion rules for third party audit on fips

Closes #33179

### CODE REVIEW

**Review by jkakavas — APPROVED** [review]
Added a couple of suggestions for comments as what we're doing here might not be obvious, but other than that LGTM

File: distribution/tools/plugin-cli/build.gradle [hunk 1]
```diff
@@ -39,3 +39,7 @@ test {
   // TODO: find a way to add permissions for the tests in this module
   systemProperty 'tests.security.manager', 'false'
 }
+
+if (project.inFipsJvm) {
+  thirdPartyAudit.enabled = false
```
Comment (jaymode) [inline distribution/tools/plugin-cli/build.gradle]: I think this definitely warrants a comment as to why we disable third party audit

**Review by jaymode — COMMENTED** [review]
I agree with @jkakavas about adding comments for why we disable things with FIPS

File: distribution/tools/plugin-cli/build.gradle (hunk 1 above)
Comment (jasontedor) [reply distribution/tools/plugin-cli/build.gradle]: I don't understand why we are completely disabling this as opposed to excluding the forbidden classes when using a FIPS JVM?

Comment (atorok) [reply distribution/tools/plugin-cli/build.gradle]: @jasontedor we could exclude but it's a _really_ long list, and those classes are AFAIK the only difference between fips and java 8 from a third party audit perspective, if we except all I don't think there's any reason to keep it enabled.

**Pull request merged**